main.go

```diff
@@ -185,3 +217,69 @@ func setupWebhooks(mgr ctrl.Manager) {
 func concurrency(c int) controller.Options {
 	return controller.Options{MaxConcurrentReconciles: c}
 }
+
+// GetTLSOptionOverrideFuncs returns a list of TLS configuration overrides to be used
+// by the webhook server.
+func GetTLSOptionOverrideFuncs(options TLSOptions) ([]func(*tls.Config), error) {
+	var tlsOptions []func(config *tls.Config)
+	var TLSMinVersionIndex, TLSMaxVersionIndex int
+	for tlsVersionIndex, TLSVersion := range cliflag.TLSPossibleVersions() {
+		if options.TLSMinVersion == TLSVersion {
+			TLSMinVersionIndex = tlsVersionIndex
+		}
+		if options.TLSMaxVersion == TLSVersion {
+			TLSMaxVersionIndex = tlsVersionIndex
+		}
+	}
+	// Check if tls min version flag value is greater than max version
+	if TLSMinVersionIndex > TLSMaxVersionIndex &&
+		TLSMaxVersionIndex != -1 &&
+		TLSMinVersionIndex != -1 {
+		return nil, fmt.Errorf("TLS version flag min version is greater than max version")
+	}
+
+	tlsMinVersion, err := cliflag.TLSVersion(options.TLSMinVersion)
+	if err != nil {
+		return nil, err
+	}
+	tlsOptions = append(tlsOptions, func(cfg *tls.Config) {
+		cfg.MinVersion = tlsMinVersion
+	})
+
+	tlsMaxVersion, err := cliflag.TLSVersion(options.TLSMaxVersion)
+	if err != nil {
+		return nil, err
+	}
+	tlsOptions = append(tlsOptions, func(cfg *tls.Config) {
+		cfg.MaxVersion = tlsMaxVersion
+	})
+	// Cipher suites should not be set if empty.
+	if options.TLSMinVersion == "VersionTLS13" &&
+		options.TLSMaxVersion == "VersionTLS13" &&
+		options.TLSCipherSuites != "" {
+		setupLog.Info("warning: Cipher suites should not be set for TLS version 1.3")
```

Comment:
I would clarify here that we are ignoring the ciphers.
```suggestion
		setupLog.Info("warning: Cipher suites should not be set for TLS version 1.3. Ignoring ciphers")
```